Android/app/src/main/java/aschi2403/tsiy/model/GPSPoint.kt

```diff
@@ -0,0 +1,44 @@
+package aschi2403.tsiy.model
+
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import androidx.room.PrimaryKey
+
+@Entity(
+    tableName = "GPSPoints",
+    indices = [Index("workoutEntryId")]
+)
+class GPSPoint(
+    @PrimaryKey(autoGenerate = true)
+    var id: Long? = null,
+    var latitude: Double,
+    var longitude: Double,
+    var workoutEntryId: Long
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as GPSPoint
+
+        if (id != other.id) return false
```

Comment:
please checkout when statement: 
https://kotlinlang.org/docs/control-flow.html#when-expression 
possible usage:
return when {
            id != other.id -> false
            else -> true
        }

Reply:
This is just a automatic generated method?